addons/kafka/scripts/kafka-exporter-setup.tpl

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+{{- $clusterName := $.cluster.metadata.name }}
+{{- $namespace := $.cluster.metadata.namespace }}
+{{- /* find kafka-server component */}}
+{{- $component := fromJson "{}" }}
+{{- range $i, $e := $.cluster.spec.componentSpecs }}
+  {{- if contains "kafka-combine-server" $e.componentDef }}
+  {{- $component = $e }}
+  {{- end }}
+{{- end }}
+{{- if not $component  }}
+{{- /* find kafka-broker component */}}
+  {{- range $i, $e := $.cluster.spec.componentSpecs }}
+    {{- if contains "kafka-broker" $e.componentDef }}
+    {{- $component = $e }}
+    {{- end }}
+  {{- end }}
+{{- end }}
+{{- /* build --kafka.server= string */}}
+{{- $replicas := $component.replicas | int }}
+{{- $servers := "" }}
+{{- range $i, $e := until $replicas }}
+  {{- $podFQDN := printf "%s-%s-%d.%s-%s-headless.%s.svc.%s" $clusterName $component.name $i $clusterName $component.name $namespace $.clusterDomain }}
```

Comment:
If it is implemented based on KubeBlocks version 0.8, for such references, try to use Vars on the ComponentDefinition as much as possible. There are some built-in ENVs and other types of references available. Try to avoid referencing via the .component method.

Reply:
here will ref the broker component headlessService and the broker Replicas to build a param, no found these ENV can be used. only can ref the broker headless service is not enough